src/hitchhiker/tree/core.clj

```diff
@@ -102,11 +174,13 @@
   [children]
   (mapv last-key (butlast children)))
 
+(declare ->IndexNode)
+
 (defrecord IndexNode [children storage-addr op-buf cfg]
   IResolve
   (index? [this] true)
-  (dirty? [this] (not (realized? storage-addr)))
-  (resolve [this] this) ;;TODO this is a hack for testing
+  (dirty? [this] (not (async/poll! storage-addr)))
+  (resolve [this] (go this)) ;;TODO this is a hack for testing
```

Comment:
What is this hack helping for testing? I'd like to understand, as maybe I could refactor to make the system more testable.